changes kvtree_print to use a more compact form

adammoody opened the pull request:
Not sure about this yet, but this changes the kvtree_print output formatting from:
```
  VERSION
    2
  CURRENT
    ckpt.6
  NAME
    ckpt.6
      DSET
        6
  DSET
    6
      FLUSHED
        2020-05-13T12:36:18
      COMPLETE
        1
      DSET
        ID
          6
        NAME
          ckpt.6
        FLAG_CKPT
          1
        FLAG_OUTPUT
          0
        CREATED
          1589398578343491
        USER
          user1
        JOBNAME
          testing_job
        JOBID
          106817
        CKPT
          6
        FILES
          4
        SIZE
          2097186
        COMPLETE
          1
```
to a more compact form with key=value style for leaf elements:
```
  CURRENT = ckpt.6
  VERSION = 1
  NAME
    ckpt.6
      DSET = 6
  DSET
    6
      FLUSHED = 2020-05-13T11:37:36
      COMPLETE = 1
      DSET
        ID = 6
        NAME = ckpt.6
        FLAG_CKPT = 1
        FLAG_OUTPUT = 0
        CREATED = 1589395056752247
        USER = user1
        JOBNAME = testing_job
        JOBID = 106817
        CKPT = 6
        FILES = 4
        SIZE = 2097186
        COMPLETE = 1
```
May be nice to make this optional and provide both forms.

CamStan commented:
Not a lot of AXL tests, but this managed to make one fail:
```
test 2
    Start 2: print_test

2: Test command: /home/travis/build/ECP-VeloC/KVTree/test/print_test "/home/travis/build/ECP-VeloC/KVTree/build/src/kvtree_print_file" "/home/travis/build/ECP-VeloC/KVTree/test/files"
2: Test timeout computed to be: 9.99988e+06
2/6 Test #2: print_test .......................***Failed    0.02 sec
```

Essentially this tests calls `kvtree_print_file` on [flush.scr](https://github.com/ECP-VeloC/KVTree/blob/master/test/files/flush.scr) and expects the result to match [flush_file_print.scr](https://github.com/ECP-VeloC/KVTree/blob/master/test/files/flush_file_print.scr).

adammoody commented:
Added a second ```kvtree_print_mode``` function that takes a mode argument to specify the format to be used while printing.  One can use either ```KVTREE_PRINT_TREE``` or ```KVTREE_PRINT_KEYVAL```.

For backwards compatibility, the ```kvtree_print``` function now just calls this new function with a hard-coded ```KVTREE_PRINT_TREE```.

I renamed the ```kvtree_print_file``` binary to ```kvtree_print``` and added a new ```--mode tree|keyval``` option.

adammoody commented:
Thanks @CamStan and @tonyhutter
